src/app/auth/auth.component.ts

```diff
@@ -1,15 +1,35 @@
 import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+import { User } from './auth';
+import { AuthService } from './auth.service';
+import { LocalstorageService } from '../services/localstorage.service';
 
 @Component({
   selector: 'app-auth',
   templateUrl: './auth.component.html',
-  styleUrls: ['./auth.component.scss']
+  styleUrls: ['./auth.component.scss'],
 })
 export class AuthComponent implements OnInit {
+  userNameToAuthorize: string = '';
+  isAuthorizing: boolean = false;
 
-  constructor() { }
+  constructor(
+    private localstorageService: LocalstorageService,
+    private authService: AuthService,
+    private router: Router
+  ) {}
 
-  ngOnInit(): void {
-  }
+  ngOnInit(): void {}
 
+  getToken(): void {
```

Comment:
this logic should be inside auth.service -> method should be called login() and use the other method from AuthService